app/controllers/mixins/generic_feature_action_mixin.rb

```diff
@@ -0,0 +1,19 @@
+module Mixins
+  module GenericFeatureActionMixin
```

Comment:
I think this is a little over-engineered, what about just defining the `model_feature_for_action` on the two affected controllers? It's definitely less lines than this mixin and it would not have that big conditional complexity.

Reply:
@skateman you may be right about the over-engineering, but  tried to avoid from duplicating the code in 3 controllers (storage, block storage and object storage). for now this is only the edit function, but maybe in the future there will be additional features in it, so the duplicate code can be (in potencial) bigger. both options are not optimal in my opinion, I will go with whatever you suggest

Reply:
Do you really need this in object storage for now?

Reply:
yes, because object storage and block storage are not a standalone manager but a mixin inside ems_storage they dont have a ems_block_storage_center or ems_object_storage_center and they use ems_storage_center so both of them will get the edit button functionallity so we need to handle it also for object storage

Reply:
I never liked this semi-inheritance between these controllers :confused: let's just define the method in each, I don't see the point in adding the extra complexity. If you really want to avoid repeating code, create a `StorageCommonMixin` and define just the `model_feature_for_action` method there. We need to keep a mixin anyway for having the common methods somewhere...

Reply:
ok, I also think we should use a separate manager for each, but this was decided in a meeting with adam and jason a few months ago. I do think this is something we should consider in the future again (replace the mixin with a manager)

Reply:
@skateman I will do the changes you suggested until tomorrow morning and inform you about it

Reply:
@skateman done. I like the result with StorageCommonMixin, waiting for your feedback